internal/proposals/migration-hooks/migration-hooks-proposal.md

```diff
@@ -0,0 +1,233 @@
+# JobManager (Migration Logic)
+
+This PoC is used to come up with a valid design for the new __JobManager__, needed to have a fully automated Kyma deploy possible. It will be used to configure the cluster and the components during the deployment of Kyma. The term `Deployment`/`Deploy` is being used in the context of installing Kyma on an empty cluster or to upgrade Kyma from an older to a newer version.
+
+To achieve a valid solution for the PoC we need to come up with a design for the following:
+
+- [Work-Flow / General mechanism](#work-flow--general-mechanism)
+  - [Requirements](#requirements)
+  - [Possible Solution](#possible-solution)
+- [Draft for Golang Implementation](#draft-for-golang-implementation)
+- [Placement of logic and actual jobs](#placement-of-logic-and-actual-jobs)
+- [Additions](#additions)
+
+## Work-Flow / General mechanism
+
+### Requirements
+
+- Only support single linear upgrade: A &#8594; B && B &#8594; C; NOT A &#8594; C. This is due to the fact that Kyma only supports single linear upgrades.
+- Mechanism should only trigger for one specific Kyma version, this could be: The Kyma version which wants to be installed on an empty cluster, Kyma version to upgrade to, or the Kyma version which should be uninstalled from the cluster.
+- This mechanism supports jobs for two different main use-cases: The __component based__ jobs and the __global/non-component based__ jobs
+  - __Component based__:
+    - Check if component is installed on cluster, or if it wants to be newly installed, and only trigger if yes
+    - It should be possible to trigger jobs before and after an deployment of a component
+  - __global / Component independent__:
+    - Always trigger logic when installing, upgrading, or uninstalling Kyma
+    - It should be possible to trigger jobs before and after the deployment, deletion of Kyma
+    - We call them `global` jobs to stick to the naming convention of our helm charts
```

Comment:
Is this a requirement or just a statement why we call them that?
If it's a requirement, I'd suggest:

```suggestion
    - Call component-independent jobs `global` jobs to stick to the naming convention of our helm charts.
```